src/utils/error.ts

```diff
@@ -1,9 +1,63 @@
-import { BeeError } from './types'
+import { getFieldOrNull } from '.'
+import { printer } from '../printer'
 
-export function isNotFoundError(error: BeeError): boolean {
-  return error.status === 404
+export interface BeeErrorOptions {
+  notFoundMessage?: string
 }
 
-export function isInternalServerError(error: BeeError): boolean {
-  return error.status === 500
+function hasStatusCode(error: unknown, statusCode: number): boolean {
+  return getFieldOrNull(error, 'status') === statusCode
+}
+
+function isNotFoundError(error: unknown): boolean {
+  return hasStatusCode(error, 404)
+}
+
+function isInternalServerError(error: unknown): boolean {
+  return hasStatusCode(error, 500)
+}
+
+export function handleError(error: unknown, options?: BeeErrorOptions): void {
+  const message = getFieldOrNull(error, 'message')
```

Comment:
can't it be like this? as I checked you mainly using either this or that to print error message.
```suggestion
  const message = typeof error === 'string' ? error : getFieldOrNull(error, 'message')
```